packages/ethers/src/client.ts

```diff
@@ -190,12 +190,14 @@ export class Web3Modal extends Web3ModalScaffold {
           onUri(uri)
         })
 
-        if (siweConfig?.options?.enabled) {
+        const params = await siweConfig?.getMessageParams?.()
+        // Must perform these checks to satify optional types
+        if (siweConfig?.options?.enabled && params && Object.keys(params || {}).length > 0) {
```

Comment:
are the params required for the `authenticate` call?

Reply:
yes they are required. Idea is, if the message params are empty, it fallsback to legacy connect + sign